src/transports/dash/manifest_parser.ts

```diff
@@ -163,20 +164,30 @@ export default function generateManifestParser(
             if (typeof resource.responseData !== "string") {
               throw new Error("External DASH resources should have been a string");
             }
-            // Normally not needed but TypeScript is just dumb here
-            return objectAssign(resource, { responseData: resource.responseData });
+            return objectAssign(resource, {
+              responseData: { data: resource.responseData, success: true as const },
+            });
           });
           return processMpdParserResponse(value.continue(resources));
         } else {
           const resources = loadedResources.map(resource => {
             if (!(resource.responseData instanceof ArrayBuffer)) {
               throw new Error("External DASH resources should have been ArrayBuffers");
             }
-            // Normally not needed but TypeScript is just dumb here
-            return objectAssign(resource, { responseData: resource.responseData });
+            return objectAssign(resource, {
+              responseData: { data: resource.responseData, success: true as const },
+            });
           });
           return processMpdParserResponse(value.continue(resources));
         }
+      }).catch((err) => {
+        const error = formatError(err, {
+          defaultCode: "PIPELINE_PARSE_ERROR",
+          defaultReason: "An unknown error occured when parsing ressources.",
+        });
+        return processMpdParserResponse(value.continue([
```

Comment:
I don't fully follow the logic here.

If `value.continue` (L181) throws because of a parsing error, then we retry calling it without a resource?

Reply:
I'm wondering if the ManifestFetcher (`src/core/fetchers/manifest/manifest_fetcher.ts`) doesn't already take care of that.

If `parseManifest` throws, we seem to go to the `emitError` function, which should call `formatError`. I didn't test it but it looks likely